src/lib/Preview.js

```diff
@@ -945,6 +996,30 @@ class Preview extends EventEmitter {
         this.viewer.addListener('viewerevent', this.handleViewerEvents);
     }
 
+    /**
+     * Handles log events and delegates to the Logger instance.
+     *
+     * @param {Object} data - Log event data.
+     * @return {void}
+     */
+    handleLogEvent(data) {
```

Comment:
This is what I mentioned in the description of the PR. Do we want viewers emitting these messages, and Preview handles it, or do we want viewers to have a `LogLayer` instance reference, and call methods on it, directly?

Reply:
Assuming Preview is not doing anything useful with the information (just passing it to logLayer), we might as well just pass the instance into the viewers and let them log from there. Going through preview would just create another route that adds no functionality to get to the same place.

Reply:
On the other hand, we could solve this third party logging problem by just broadcasting events and letting devs consume the events and do what they want with them.